documentation/query-language.html

```diff
@@ -329,6 +329,58 @@ <h3 id="near-and-ordered-near">Near and Ordered Near</h3>
 </p>
 
 
+<h3 id="sameElement">Search within same struct element</h3>
+<p>
+<em>sameElement()</em> is an operator that requires the terms to match within the same struct element in an array or a map field. Example:
+</p>
+<pre>
+...
+struct person {
+  field first_name type string {}
+  field last_name type string {}
+  field year_of_birth type int {}
+}
+
+field persons type array&lt;person&gt; {
+  indexing: summary
+  struct-field first_name { indexing: attribute }
+  struct-field last_name { indexing: attribute }
+  struct-field year_of_birth { indexing: attribute }
+}
+field identities type map&lt;string, person&gt; {
+  indexing: summary
+  struct-field key { indexing: attribute }
+  struct-field value.first_name { indexing: attribute }
+  struct-field value.last_name { indexing: attribute }
+  struct-field value.year_of_birth { indexing: attribute }
+}
+...
+</pre>
+<p>
+With normal <em>AND</em> the query <pre>persons.first_name AND persons.last_name</pre> will normally not give you what you want.
+It will match if a document has a <em>persons</em> element with a matching <em>first_name</em>
+<em>AND</em> any element with a matching <em>last_name</em>.
+So you will get a lot of false positives since there is nothing limiting them to the same element.
+However, that is what <em>sameElement</em> ensures.
+</p>
+<pre class="urlunencode" oncopy="">
+select%20%2A%20from%20sources%20%2A%20where%20persons%20contains%20sameElement%28first_name%20contains%20'Joe',%20last_name%20contains%20'Smith',%20year_of_birth%20&lt;%201940%29%3B
```

Comment:
Not entirely sure how the `urlunencode` magic class works, but assuming this is meant to be copy&paste-able I'd suggest replacing `&lt;` here with `%3C`. The former is HTML escaping, the latter is URI escaping.